src/misc/fs_utils.cpp

```diff
@@ -169,3 +169,59 @@ int64_t get_native_file_position(const NativeFileHandle handle)
 {
 	return seek_native_file(handle, 0, NativeSeek::Current);
 }
+
+std::optional<uint8_t> read_u8_at(std::ifstream &stream, std::ifstream::pos_type pos)
+{
+	stream.seekg(pos);
+	if (stream.fail()) {
+		return {};
+	}
+	uint8_t ret = 0;
+	stream.read(reinterpret_cast<char *>(&ret), sizeof(uint8_t));
+	if (stream.fail()) {
+		return {};
+	}
+	return ret;
+}
+
+std::optional<uint16_t> read_native_u16_at(std::ifstream &stream, std::ifstream::pos_type pos)
```

Comment:
Confused about the name `read_native_u16_at`. Shouldn't this be called `read_le16_at()`? That would mean it treats the input as little-endian, and of course the output is native byte order (that's implied).

If you want to be explicit, then `read_le16_as_native_u16_at`.